pkg/k8s/informers/informers.go

```diff
@@ -207,6 +164,36 @@ func (ic *InformerCollection) run(stop <-chan struct{}) error {
 	return nil
 }
 
+// Add is only exported for the sake of tests and requires a testing.T to ensure it's
+// never used in production
+func (ic *InformerCollection) Add(key InformerKey, obj interface{}, t *testing.T) error {
```

Comment:
Can you add the comment about the delay between adding to the k8s client, and checking its existence here?